src/axom/quest/SamplingShaper.hpp

```diff
@@ -275,6 +316,38 @@ class SamplingShaper : public Shaper
     m_volfracOrder = volfracOrder;
   }
 
+  /// Registers a function to project from 2D input points to 2D query points
+  template <int FromDim, int ToDim>
+  std::enable_if_t<FromDim == 2 && ToDim == 2, void> setPointProjector(
+    shaping::PointProjector<2, 2> projector)
```

Comment:
Note: these `enable_if`s are necessary. The compiler complained when I tried the simpler form:
```cpp
  void setPointProjector(shaping::PointProjector<2, 2> projector)
```

Reply:
Was it a compilation error or warning? Which compiler and what did it say?

Is the issue that this is only correct when FromDim == ToDim? Should that be handled with a static assert?

Reply:
Errors in clang@10 when calling `setPointProjector<3,2>(...)` in the shaping driver example.
The messages were related to it not finding an appropriate match in `setPointProjector<2,2>` for the 3D `FromDim`.

I was thinking about it some more, and the error might have been that I didn't remove the template params (`<3,2>`) from the calling site. I'll dig into it some more and see if I can simplify this.

Reply:
**Update:** The error _was_ at the calling site and I was able to remove the `enable_if`s here. I checked that I am not able to remove the `enable_if`s from the `sampleInOutField` function.

Thanks for poking at this @rhornung67!